File: MovieTicketBooking/Controllers/BookingController.cs
using MovieTicketBooking.Model;
using MovieTicketBooking.Services;

namespace MovieTicketBooking.Controllers
{
    public class BookingController
    {
        ShowService _showService;
        TheatreService _theatreService;
        BookingService _bookingService;
        public BookingController(ShowService showService,
        TheatreService theatreService, BookingService bookingService)
        {
            _showService = showService;
            _theatreService = theatreService;
            _bookingService = bookingService;
        }
        public string CreateBooking(string showId, string user, List<string> seatIds)
        {
            Show show = _showService.GetShow(showId);
            List<Seat> seats = seatIds.Select(_theatreService.GetSeat).ToList();
            return _bookingService.CreateBooking(show, user, seats);
        }
        public void ConfrimBooking(string bookingId, string user)
        {
            _bookingService.ConfrimBooking(bookingId, user);
        }
    }
}

File: MovieTicketBooking/Movie Booking Java/src/main/java/org/example/controllers/BookingController.java
package org.example.controllers;

import lombok.AllArgsConstructor;
import lombok.NonNull;
import org.example.model.Booking;
import org.example.model.Seat;
import org.example.model.Show;
import org.example.service.BookingService;
import org.example.service.ShowService;
import org.example.service.TheaterService;

import java.util.List;

@AllArgsConstructor
public class BookingController {

    private final BookingService bookingService;
    private final ShowService showService;
    private final TheaterService theaterService;

    public Booking createBooking(@NonNull final String showId,
                                 @NonNull final String userId,
                                 @NonNull final List<String> seatIds) {

        List<Seat> seats = seatIds.stream().map(theaterService::getSeat).toList();
        Show show = showService.getShow(showId);
        return bookingService.createBooking(userId, show, seats);
    }
}
